### DISCUSSION TITLE
Using secrets in python test with tox

### DISCUSSION THREAD

**Discussion by romanjergon** [community/community#26745] [Actions]
<p>Hello everyone,</p>
<p>I am trying to use github actions and secrets in my python tests which I run via tox in repo <a>https://github.com/romanjergon/flight_deals_searcher</a>. Unfortunately I am not able to get the github secret value to python script even though it seems everything works in my other repo (without use of tox).</p>
<p>I have defined .yml file in .github/workflows/tests.yml</p>
<pre><code class="lang-auto">name: Tests

on:
  - push
  - pull_request

jobs:
  test:
    runs-on: ${{ matrix.os }}
    strategy:
      matrix:
        os: [ubuntu-latest, windows-latest]
        python-version: ['3.9', '3.10']

    steps:
    - uses: actions/checkout@v2
    - name: Set up Python ${{ matrix.python-version }}
      uses: actions/setup-python@v2
      with:
        python-version: ${{ matrix.python-version }}
    - name: Install dependencies
      run: |
        python -m pip install --upgrade pip
        pip install tox tox-gh-actions
    - name: Test with tox
      run: tox
      env:
        KIWI_API_KEY: ${{ secrets.KIWI_API_KEY }}
</code></pre>
<p>Secret is defined for the repo:<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/2/e/2e3195c4f5affc661fdd891f2175dbbd9f902120.png" data-download-href="/uploads/short-url/6AEclj7bVa3VDqsLRdovfP6Q3mw.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/93288153/181099706-981e6d48-7470-4777-bac6-87e3f58f3483.png" alt="image" data-base62-sha1="6AEclj7bVa3VDqsLRdovfP6Q3mw" width="690" height="441" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/2/e/2e3195c4f5affc661fdd891f2175dbbd9f902120_2_690x441.png, https://aws1.discourse-cdn.com/github/optimized/3X/2/e/2e3195c4f5affc661fdd891f2175dbbd9f902120_2_1035x661.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/2/e/2e3195c4f5affc661fdd891f2175dbbd9f902120.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/2/e/2e3195c4f5affc661fdd891f2175dbbd9f902120_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">1301×832 41.7 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>I have defined no environments in github.</p>
<p>Env value is used in python test script<br>
tests/test_flight_search.py</p>
<pre><code class="lang-auto">import pytest
import os
from flight_deals_searcher.flight_search import FlightSearch
from dotenv import load_dotenv

load_dotenv()

def test_flight_search():
    flight_searcher = FlightSearch(
        departure_code="PRG",
        kiwi_api_key=os.environ["KIWI_API_KEY"],
        kiwi_endpoint="http://tequila-api.kiwi.com",
        search_period_start=1,
        search_period_end=180,
    )
</code></pre>
<p>My job fails stating that the env variable does not exist - <a href="https://github.com/romanjergon/flight_deals_searcher/runs/4322925725?check_suite_focus=true" class="inline-onebox" rel="noopener nofollow ugc">testing github secrets · romanjergon/flight_deals_searcher@29371c4 · GitHub</a></p>
<p>Not sure if it plays any role but my tox configuration is in tox.ini</p>
<p>Would anyone spot what is the cause of the issue?<br>
Thank you!</p>

**Comment by romanjergon** [comment, marked as answer]
<p>As this post was originally hidden by I assume spam protection I have found solution.<br>
My tox.ini lacked definition of which env variables should be passed to the tests, tox does not pass all the variables.<br>
So I have extended [testenv] definition with</p>
<pre><code class="lang-auto">passenv =
    KIWI_API_KEY
</code></pre>
<p>which solved the issue.</p>

**Reply by mikebgrep** [reply]
You save the day with this answer.
